gen/generate.go

```diff
@@ -79,28 +113,43 @@ func GenerateGenesisFiles(genesisData GenesisData, consensus config.ConsensusPro
 	}
 
 	if sum != TotalMoney {
-		panic(fmt.Sprintf("Amounts don't add up to TotalMoney - off by %v", int64(TotalMoney)-int64(sum)))
-	}
-
-	// Backwards compatibility with older genesis files: if the consensus
-	// protocol version is not specified, default to V0.
-	proto := genesisData.ConsensusProtocol
-	if proto == protocol.ConsensusVersion("") {
-		proto = protocol.ConsensusCurrentVersion
+		fsum := float64(sum)
+		ftot := float64(TotalMoney)
+		if (math.Abs((fsum-ftot)/ftot) < 0.01) && (u64absDiff(sum, TotalMoney) < 10000) {
+			if verbose {
+				fmt.Fprintf(verboseOutWriter, "doing roundoff fixup expected total money %d actual sum %d\n", TotalMoney, sum)
+			}
+			// wallet stake is a float and roundoff might happen but we might be close enough to do fixup
+			i := 0
+			for sum != TotalMoney {
+				if sum < TotalMoney {
+					allocation[i].Stake++
+					sum++
+				} else {
+					if allocation[i].Stake > consensusParams.MinBalance {
+						allocation[i].Stake--
+						sum--
+					}
+				}
+				i = (i + 1) % len(allocation)
+			}
+		} else {
+			panic(fmt.Sprintf("Amounts don't add up to TotalMoney - off by %v", int64(TotalMoney)-int64(sum)))
```

Comment:
what kinds of values would lead to this panic? one of the inputs to this func() is genesis data which could be created from netgoal generate with specified parameters, should I generate an another genesis file if I reach this panic line?